logs_processor.go

```diff
@@ -0,0 +1,239 @@
+package datadog
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+const (
+	ArithmeticProcessorType = "arithmetic-processor"
+	AttributeRemapperType   = "attribute-remapper"
+	CategoryProcessorType   = "category-processor"
+	DateRemapperType        = "date-remapper"
+	GrokParserType          = "grok-parser"
+	MessageRemapperType     = "message-remapper"
+	NestedPipelineType      = "pipeline"
+	ServiceRemapperType     = "service-remapper"
+	StatusRemapperType      = "status-remapper"
+	TraceIdRemapperType     = "trace-id-remapper"
+	UrlParserType           = "url-parser"
+	UserAgentParserType     = "user-agent-parser"
+)
+
+// LogsProcessor struct represents the processor object from Config API.
+type LogsProcessor struct {
+	Name       *string     `json:"name"`
```

Comment:
why do we need points to primitive types ?

Reply:
I see a couple of things from making the fields all int (as they are pointers):
1. We don't need to copy over the values (e.g. string length can be arbitrary.)
2. The accessors are auto generated from a script (check-code-generation-ran.sh), it could be simpler to expect fields have predictable type.